File: src/main/java/com/university/services/impl/CityServiceImpl.java
package com.university.services.impl;

import com.university.entities.City;
import org.springframework.beans.factory.annotation.Autowired;
import com.university.repositories.CityRepository;
import com.university.services.CityService;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CityServiceImpl implements CityService {

    private final CityRepository cityRepository;

    @Autowired
    public CityServiceImpl(CityRepository cityRepository) {
        this.cityRepository = cityRepository;
    }

    @Override
    public void addCity(City city) {
        cityRepository.saveAndFlush(city);
    }

    @Override
    public void removeCity(int id) {
        cityRepository.delete(id);
    }

    @Override
    public City getCity(int id) {
        return cityRepository.findOne(id);
    }

    @Override
    public List<City> getAllCities() {
        return cityRepository.findAll();
    }
}

File: src/main/kotlin/com/university/services/impl/CityServiceImpl.kt
package com.university.services.impl

import com.university.entities.City
import com.university.repositories.CityRepository
import com.university.services.CityService
import org.springframework.stereotype.Service

@Service
class CityServiceImpl (private val cityRepository: CityRepository) : CityService {
    override fun getAll(): List<City> {
        return cityRepository.findAll()
    }
}
